test/napi/napi.test.ts

```diff
@@ -0,0 +1,64 @@
+import { it, expect, test, beforeAll, describe } from "bun:test";
+import { spawnSync } from "bun";
+import { join } from "path";
+
+describe("napi", () => {
+  beforeAll(() => {
+    // build gyp
+    const build = spawnSync({
+      cmd: ["yarn", "build"],
+      cwd: join(__dirname, "napi-app"),
+    });
+    if (!build.success) {
+      console.error(build.stderr.toString());
+      throw new Error("build failed");
+    }
+  });
+
+  describe("napi_get_value_string_utf8 with buffer", () => {
+    // see https://github.com/oven-sh/bun/issues/6949
+    it("copies one char", () => {
+      const result = checkSameOutput("test_napi_get_value_string_utf8_with_buffer", ["abcdef", 2]);
+      expect(result).toEndWith("str: a");
+    });
+
+    it("copies null terminator", () => {
+      const result = checkSameOutput("test_napi_get_value_string_utf8_with_buffer", ["abcdef", 1]);
+      expect(result).toEndWith("str:");
+    });
+
+    it("copies zero char", () => {
+      const result = checkSameOutput("test_napi_get_value_string_utf8_with_buffer", ["abcdef", 0]);
+      expect(result).toEndWith("str: ******************************");
+    });
+
+    it("copies more than given len", () => {
+      const result = checkSameOutput("test_napi_get_value_string_utf8_with_buffer", ["abcdef", 25]);
+      expect(result).toEndWith("str: abcdef");
+    });
+
+    it("copies auto len", () => {
+      const result = checkSameOutput("test_napi_get_value_string_utf8_with_buffer", ["abcdef", 424242]);
+      expect(result).toEndWith("str:");
+    });
+  });
+});
+
+function checkSameOutput(test: string, args: any[]) {
+  const nodeResult = runOn("node", test, args).trim();
+  let bunResult = runOn(join(__dirname, "../../build/bun-debug"), test, args);
```

Comment:
```suggestion
  let bunResult = runOn(bunExe(), test, args);
```

Reply:
If I do this locally, tests fail again as if it was run on the current bun version... is `bunExe()` supposed to refer to the newly built version of bun ?

Reply:
hm. yes, is `bun-debug` available in `$PATH`?

Reply:
yes, but it's available as `bug-debug`, not `bun`. Thus, these tests only pass locally if I run them via `bun run build && bun-debug test test/napi`  (with a lot of noisy logs, obviously).

If that's the executable that runs on your test pipeline instead of the current version of bun, then fine, lets commit that 🙃

Reply:
The tests pass locally with `process.execPath` in use and they will not pass in CI without that.